### ISSUE TITLE
Zoom out when returning to GPS position 

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#828]
## Bug/Feature
(Delete the non-relevant)

## What I expect to happen
The app keeps the zoom level when returning to GPS position 

## What really happened

When GPS tracking or recording is active, fly to GPS position may not preserve the pre-flight zoom level. Instead, the map zooms out. 

## What I did that caused the issue - step by step
1. Perform the following steps while driving using the app or the site
2. Start GPS tracking 
3. Pan and to a far location like Eilat and zoom in to steet level.
4. Return to GPS position
The final zoom level is lower - "50km" scale

## My environment
* Operating System and version: Android 8.0.0
* Web Browser and version: 7.2.63 (app and web site) 

## Thing I think the developers should know, images, links etc.
Perhaps the position update is resumed before the fly-to completes the final zoom-in and accidentally stops it.

**Comment by zstadler** [comment]
I see two possible directions for addressing this issue, and there may be additional directions.

1. Always provide the zoom level when [`setLocation`](https://github.com/IsraelHikingMap/Site/blob/1032658b7dd9f9150b2f9fd5002c56d8bad456de/IsraelHiking.Web/sources/application/components/location.component.ts#L274-L276) is calling [`flyTo`](https://github.com/IsraelHikingMap/Site/blob/8b6df7f5a62e6c4233c3b7a4139f2ec6cbbfa70a/IsraelHiking.Web/sources/application/services/fit-bounds.service.ts#L32-L36). This will ensure that the position update will perform the remaining zoom update, if needed.

2. Capture such interruption and handle it.  According to the [`view.animate` documentation](https://openlayers.org/en/latest/apidoc/module-ol_View-View.html)

   > If you provide a function as the last argument to the animate method, it will get called at the end of an animation series. The callback will be called with true if the animation series completed on its own or false if it was cancelled.
   >
   >Animations are cancelled by user interactions (e.g. dragging the map) or by calling view.setCenter(), view.setResolution(), or view.setRotation() (or another method that calls one of these).

Although options 1 may seem simpler, `zoom` and `location` are independent components.

**Comment by HarelM** [comment]
I don't think the `setlocation` should keep the zoom level since the user can change it, and maintaining the current zoom level in the location component is just asking for bugs - understanding if the zoom change is due to a user interaction, a search, a poi opening etc is almost impossible.
I'm in favor of closing this bug. @zstadler please approve.

**Comment by zstadler** [comment]
Do you agree with me about the desired user-experience and the problem with it is the implementation complexity?

**Comment by HarelM** [comment]
I agree with the general concept of returning to the same zoom level "the user left". The problem besides the implementation (which is a big problem as far as I can understand) is the definition of the "the user left" part of the requirement.
It seems to me that there are a lot of cases that needs to be defined how the application should behave, defining them and implementing them seems like a great task which the value is low, as the user can simply zoom to the required level.
If the zoom was automatic like it is in Waze I think it would've been simpler, but I don't think an automatic zoom is what we want.
To summarize, I think the implementation and definition is too complicated.

**Comment by zstadler** [comment]
I'll take the task of creating a high-level design that will attempt to be closer to the desired behavior while keeping the implementation complexity low.

**Comment by HarelM** [comment]
Assumed to have been fixed as part of: #1014, commit: 941ec92b8d75ea0064516f4b98f912bfbf5f57ce

**Comment by zstadler** [comment]
Indeed, this issue was resolved in 941ec92b8d75ea0064516f4b98f912bfbf5f57ce by [adding this condition](https://github.com/IsraelHikingMap/Site/commit/941ec92b8d75ea0064516f4b98f912bfbf5f57ce#diff-ade51f38307f67edc40dd1e6816e0c44R259).

Confirmed by repeating the reproduction steps.

**Comment by HarelM** [comment]
@zstadler can I delete the issue-828 branch?
